opentelekomcloud/services/rms/data_opentelekomcloud_rms_advanced_query_schemas_v1.go

```diff
@@ -0,0 +1,135 @@
+package rms
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	golangsdk "github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/rms/advanced"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/cfg"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/fmterr"
+)
+
+func DataSourceRmsAdvancedQuerySchemasV1() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceRmsAdvancedQuerySchemasRead,
+
+		Schema: map[string]*schema.Schema{
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"schemas": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"schema": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceRmsAdvancedQuerySchemasRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := common.ClientFromCtx(ctx, rmsClientV1, func() (*golangsdk.ServiceClient, error) {
+		return config.RmsV1Client(config.GetRegion(d))
+	})
+	if err != nil {
+		return fmterr.Errorf(errCreationRMSV1Client, err)
+	}
+
+	res, err := advanced.ListSchemas(client, advanced.ListSchemasOpts{
+		DomainId: GetRmsDomainId(client, config),
+	})
+	if err != nil {
+		return diag.Errorf("error getting the advanced query schemas list from server: %s", err)
+	}
+
+	schemaType := d.Get("type").(string)
+
+	// Фильтруем схемы по типу, если тип указан
```

Comment:
wow cyrillic, better to omit

Reply:
yeah I prefer chinese comments but cyrillic is nice for variety..